test/integration/targets/get_certificate/tests/validate.yml

```diff
@@ -0,0 +1,90 @@
+---
+- name: Test get_certificate
+  connection: local
+  hosts: localhost
+  gather_facts: false
+  tasks:
+
+    - name: Get red hats certificate
+      get_certificate:
+        host: www.redhat.com
+        port: 443
+      register: result
+
+    - debug: var=result
+
+    - assert:
+        that:
+          # This module should never change anything
+          - "result.changed == false"
+          - "result.failed == false"
+          # We got the correct ST from the cert
+          - "'North Carolina' == result.cert.ST"
+
+    - name: Connect to http port (will fail because there is no SSL cert to get)
+      get_certificate:
+        host: www.bbc.com
+        port: 80
+      register: result
+      ignore_errors: true
+
+    - assert:
+        that:
+          - "result.changed == false"
+          - "result.failed == true"
+          # We got the expected error message
+          - "'The handshake operation timed out' in result.msg"
+ 
+    - name: Test timeout option
+      get_certificate:
+        host: www.redhat.com
+        port: 1234
+        timeout: 1
```

Comment:
If `redhat.com`s firewall rules change to `DENY` instead of `DROP`, this may well return a result within a second.

Reply:
True. Again, if we have some kind of server defined at test time that we control we could do something different here to generate a time out I am more than willing to use that.